src/liballoc/rc.rs

```diff
@@ -782,8 +782,9 @@ impl<T: ?Sized> Drop for Weak<T> {
     fn drop(&mut self) {
         unsafe {
             let ptr = *self._ptr;
-            if !(*(&ptr as *const _ as *const *const ())).is_null() &&
-               ptr as *const () as usize != mem::POST_DROP_USIZE {
+            let thin = ptr as *const ();
+
+            if !(thin as *const *const ()).is_null() && thin as usize != mem::POST_DROP_USIZE {
```

Comment:
There's a bug open for it generally, I think https://github.com/rust-lang/rust/issues/30577